src/awkward1/operations/convert.py

```diff
@@ -1139,6 +1139,180 @@ def getfunction(layout, depth):
     else:
         return out
 
+def to_arrow(layout):
+    import pyarrow
+
+    def recurse(layout, mask = None):
+
+        if isinstance(layout, awkward1.layout.NumpyArray):
+            numpy_arr = numpy.asarray(layout)
+            if(numpy_arr.ndim == 1):
+                return pyarrow.array(numpy_arr, mask=mask)
+            else:
+                return pyarrow.Tensor.from_numpy(numpy_arr)
+        
+        elif isinstance(layout, awkward1.layout.EmptyArray):
+            return pyarrow.array(numpy.array([], dtype=numpy.float64))
+
+        elif isinstance(layout, (awkward1.layout.IndexU8, 
+                                awkward1.layout.IndexU32, 
+                                awkward1.layout.Index8, 
+                                awkward1.layout.Index32, 
+                                awkward1.layout.Index64)):
+            return pyarrow.array(numpy.asarray(layout))
+        
+        elif isinstance(layout, awkward1.layout.ListOffsetArray32):
+
+            if (awkward1.operations.describe.parameters(layout).get("__array__") == "bytestring"):
+                arr = [awkward1.behaviors.string.ByteBehavior(
+                    layout[i]).__bytes__() for i in range(len(layout))]
+                if mask is not None:
+                    return pyarrow.array(arr, mask=numpy.array(mask, dtype=numpy.bool))
+                else:
+                    return pyarrow.array(arr)
```

Comment:
I didn't check this, I mostly relied on the `ByteString` being analogues to `CharString`.

Reply:
ByteString and CharString are very similar, differing only in the interpretation of the bytes (i.e. no interpretation vs UTF-8). They correspond to `pyarrow.binary` and `pyarrow.string`.